compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/AbstractSuspendFunctionsLowering.kt

```diff
@@ -80,7 +80,10 @@ abstract class AbstractSuspendFunctionsLowering<C : CommonBackendContext>(val co
     // Suppress since it is used in native
     @Suppress("MemberVisibilityCanBePrivate")
     protected fun IrCall.isReturnIfSuspendedCall() =
-        symbol.owner.run { fqNameWhenAvailable == context.internalPackageFqn.child(Name.identifier("returnIfSuspended")) }
+        symbol.owner.run {
+            parent.getPackageFragment()?.fqName == context.internalPackageFqn &&
+                    name == Name.identifier("returnIfSuspended")
```

Comment:
```suggestion
                    name.asString() == "returnIfSuspended"
```

Reply:
BTW maybe this whole function could rather be refactored to something like `return symbol.signature == context.returnIfSuspendedSignature` where the latter is created in the context as something like `IdSignature.PublicSignature(internalPackageFqn, "returnIfSuspended", null, 0)`. (But I don't know if this works.)

In general I think we should move more backend code to dealing with `IdSignature`s where `owner` is not needed, as in this case, since it could be more performant.

Reply:
Now that I think about, maybe we could refactor the whole `irTypePredicates.kt` and similar utilities to use `IdSignature` as well, to only deal with strings and avoid `FqName` (whose `parent` is also not very cheap sometimes)

Reply:
> Now that I think about, maybe we could refactor the whole irTypePredicates.kt and similar utilities to use IdSignature as well, to only deal with strings and avoid FqName

Thank you for the idea. FqName is really quite expensive. Getting parent isn't very cheap, but in the huge number of cases it'd be called quite rare (I can told about this looking at profile results of K/N compiler). I'll try to use IdSignature a bit later and  try to measure if it's cheaper.